aggregator/src/aggregator.rs

```diff
@@ -3194,119 +3193,104 @@ impl VdafOps {
-                                    )
-                                    .await
-                                })),
-                                try_join_all(
-                                    empty_batch_aggregations
-                                        .iter()
-                                        .map(|ba| tx.put_batch_aggregation(ba))
-                                ),
-                            )?;
-                            aggregate_share_job
-                        }
-                    };
+                    let (mut helper_aggregate_share, report_count, checksum) =
+                        compute_aggregate_share::<SEED_SIZE, Q, A>(&task, &batch_aggregations)
+                            .await
+                            .map_err(|e| datastore::Error::User(e.into()))?;
 
-                    // §4.4.4.3: verify total report count and the checksum we computed against
-                    // those reported by the leader.
-                    if aggregate_share_job.report_count() != aggregate_share_req.report_count()
-                        || aggregate_share_job.checksum() != aggregate_share_req.checksum()
-                    {
-                        return Err(datastore::Error::User(
-                            Error::BatchMismatch(Box::new(BatchMismatch {
-                                task_id: *task.id(),
-                                own_checksum: *aggregate_share_job.checksum(),
-                                own_report_count: aggregate_share_job.report_count(),
-                                peer_checksum: *aggregate_share_req.checksum(),
-                                peer_report_count: aggregate_share_req.report_count(),
-                            }))
-                            .into(),
-                        ));
-                    }
+                    vdaf.add_noise_to_agg_share(
+                        &dp_strategy,
+                        &aggregation_param,
+                        &mut helper_aggregate_share,
+                        report_count.try_into()?,
+                    )
+                    .map_err(|e| datastore::Error::User(e.into()))?;
 
+                    // Now that we are satisfied that the request is serviceable, we consume
+                    // a query by recording the aggregate share request parameters and the
+                    // result.
+                    let aggregate_share_job = AggregateShareJob::<SEED_SIZE, Q, A>::new(
+                        *task.id(),
+                        aggregate_share_req
+                            .batch_selector()
+                            .batch_identifier()
+                            .clone(),
+                        aggregation_param,
+                        helper_aggregate_share,
+                        report_count,
+                        checksum,
+                    );
+                    try_join!(
```

Comment:
This should be controlled by a flag (as well as the equivalent scrubbing functionality in the collection job driver) -- but I think this isn't necessary unless/until we backport to a released version.